lib/BackgroundJobs/StorageCrawlJob.php

```diff
@@ -104,13 +146,18 @@ protected function run($argument): void {
 		try {
 			$path = $root['path'] === '' ? '' :  $root['path'] . '/';
 			$qb = new CacheQueryBuilder($this->db, $this->systemConfig, $this->logger);
-			$files = $qb->selectFileCache()
+			$ignoreFileidsChunks = array_chunk($ignoreFileids, 999, true);
+			$ignoreFileidsExpr = array_map(fn ($chunk) => $qb->expr()->notIn('parent', $qb->createNamedParameter($chunk, IQueryBuilder::PARAM_INT_ARRAY)), $ignoreFileidsChunks);
+			$qb->selectFileCache()
 				->whereStorageId($storageId)
 				->andWhere($qb->expr()->like('path', $qb->createNamedParameter($path . '%')))
 				->andWhere($qb->expr()->eq('storage', $qb->createNamedParameter($storageId)))
 				->andWhere($qb->expr()->in('mimetype', $qb->createNamedParameter($mimeTypes, IQueryBuilder::PARAM_INT_ARRAY)))
```

Comment:
Can we perhaps craft a query that combines e.g. video mime types with video ignore files etc.

so something like

```
->andWhere(
	$qb->expr->orX(
		$qb->expr()->andX(
			$qb->expr()->in('mimetype', $imageTypes),
			$qb->expr()->notIn('parent', $ignoreImageFileids)
		),
		$qb->expr()->andX(
			$qb->expr()->in('mimetype', $videoTypes),
			$qb->expr()->notIn('parent', $ignoreVideoFileids)
		)
	)
)
```

Reply:
ok.
This seems like a good way to do it.

Reply:
Modified. Can I re-review it?